src/overlays/actors/ovl_En_Kakasi2/z_en_kakasi2.c

```diff
@@ -20,21 +42,192 @@ const ActorInit En_Kakasi2_InitVars = {
     (ActorFunc)EnKakasi2_Update,
     NULL,
 };
-*/
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Kakasi2/EnKakasi2_Init.s")
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Kakasi2/EnKakasi2_Destroy.s")
+void EnKakasi2_Init(Actor* thisx, GlobalContext* globalCtx) {
+    EnKakasi2* this = THIS;
+    s32 pad;
+    f32 argument0;
+    f32 zAngle;
+
+    osSyncPrintf("\n\n");
+    // Visit Umeda
+    osSyncPrintf(VT_FGCOL(GREEN) "☆☆☆☆☆ 梅田参号見参！ ☆☆☆☆☆ \n" VT_RST);
+
+    this->saveFlag = this->actor.params & 0x3F;
+    argument0 = (this->actor.params >> 6) & 0xFF;
+    zAngle = this->actor.posRot.rot.z;
+    if (this->saveFlag == 0x3F) {
+        this->saveFlag = -1;
+    }
+    this->actor.unk_1F = 4;
+    this->position.x = (argument0 * 40.0f) + 40.0f;
+    this->position.y = (zAngle * 40.0f) + 40.0f;
+
+    // Former? (Argument 0)
+    osSyncPrintf(VT_FGCOL(YELLOW) "☆☆☆☆☆ 元？(引数０) ☆☆☆☆ %f\n" VT_RST, argument0);
+    // Former? (Z angle)
+    osSyncPrintf(VT_FGCOL(YELLOW) "☆☆☆☆☆ 元？(Ｚアングル) ☆☆ %f\n" VT_RST, zAngle);
+    // Correction coordinates X
+    osSyncPrintf(VT_FGCOL(YELLOW) "☆☆☆☆☆ 補正座標Ｘ ☆☆☆☆☆ %f\n" VT_RST, this->position.x); // 19C, 1A0, 1A4
+    // Correction coordinates Y
+    osSyncPrintf(VT_FGCOL(YELLOW) "☆☆☆☆☆ 補正座標Ｙ ☆☆☆☆☆ %f\n" VT_RST, this->position.y);
+    // Correction coordinates Z
+    osSyncPrintf(VT_FGCOL(YELLOW) "☆☆☆☆☆ 補正座標Ｚ ☆☆☆☆☆ %f\n" VT_RST, this->position.z);
+    osSyncPrintf(VT_FGCOL(YELLOW) "☆☆☆☆☆ SAVE       ☆☆☆☆☆ %d\n" VT_RST, this->saveFlag);
+    osSyncPrintf("\n\n");
+
+    this->actor.colChkInfo.mass = 0xFF;
+    this->height = 60.0f;
+    Actor_SetScale(&this->actor, 0.01f);
+    this->actor.flags |= 0x400;
+    this->unk_198 = this->actor.shape.rot.y;
+
+    if (this->saveFlag >= 0 && Flags_GetSwitch(globalCtx, this->saveFlag)) {
+        this->actor.draw = func_80A90948;
+        Collider_InitCylinder(globalCtx, &this->collider);
+        Collider_SetCylinder(globalCtx, &this->collider, &this->actor, &sCylinderInit);
+        SkelAnime_InitSV(globalCtx, &this->skelAnime, &D_060065B0, &D_06000214, NULL, NULL, 0);
+        this->actionFunc = func_80A9062C;
+    } else {
+        this->actionFunc = func_80A90264;
+        this->actor.shape.unk_08 = -8000.0f;
+    }
+}
+
+void EnKakasi2_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    EnKakasi2* this = THIS;
+
+    Collider_DestroyCylinder(globalCtx, &this->collider);
+}
+
+void func_80A90264(EnKakasi2* this, GlobalContext* globalCtx) {
+    Player* player = PLAYER;
+
+    this->unk_194++;
+
+    if ((BREG(1) != 0) && (this->actor.xzDistFromLink < this->position.x) &&
+        (fabsf(player->actor.posRot.pos.y - this->actor.posRot.pos.y) < this->position.y)) {
```

Comment:
`position` seems to be used as a distance so I'm not sure about the name

Reply:
I've changed it to distance, however in the printfs they're referenced as "correction coordinates"... so it might be better to name it as correctionCoordinates ?? what do you think?

Reply:
its used as a distance check to see if player is in range to spawn the scarecrow.
`spawnRange` or something similar would be good, not sure on the exact phrasing

Reply:
its being used as a `maxSpawnDistance` since it checks if player is less than the specified distance away.
and this lets us know what temp0 and temp1 is in init as well. not sure of the best phrasing for that though since theres some additional math on it before it gets stored

Reply:
Should I rename it to maxSpawnDistance? Any suggestion for the temps?